docs/rfcs/2022-12-17-user-profiles.md

```diff
@@ -0,0 +1,130 @@
+# User profiles
+
+## Problem
+
+Convenience of having different communication contexts in a single client without shared metadata.
+
+Currently requires changing chat database and restarting chat.
+
+Events (notifications) for databases other than current are not received.
+
+## Solution
+
+Support for multiple user profiles in a single database.
+
+Separate transport connections with the same servers not shared between profiles.
+
+## Design
+
+### API
+
+- new users are created using existing CreateActiveUser API
+
+- APIListUsers
+
+- APISetActiveUser UserId
+
+- CRUsersList {users :: [User]}
+
+- terminal APIs
+
+### Subscriptions
+
+- get connections for all users, subscribe to all - to receive events for users other than active
```

Comment:
There are two options here:
1. map subscriptions to users at the time of subscription (and creating connection), so agent doesn't store information about which subscription belongs to which user, but has some sort of session ID per user.
2. associate user ID with connection in the agent database.

Currently, I like option 1 more as it would allow to prioritise subscriptions to the currently active profile.